jupyter_client/connect.py

```diff
@@ -167,7 +167,7 @@ def write_connection_file(
     if hasattr(stat, "S_ISVTX"):
         # set the sticky bit on the file and its parent directory
         # to avoid periodic cleanup
-        paths = [fname]
+        paths = []
```

Comment:
Since we're only concerned about the hosting directory, could this be simplified to work directly against `runtime_dir`, fixup comments, etc.?

Reply:
I've created another pull request which gets rid of the for loop and sets the sticky bit only on the containing directory.

Reply:
... which had a left-over wrong comparison, now removed.